polygonizer_dockwidget.py

```diff
@@ -240,6 +241,8 @@ def eventPushButtonRunPolygonizerOnClick(self):
                     # FIXME we need some snapping flexibility, because this can leave slivers
                     if road.geometry().length() < LEG_LENGTH * 2:
                         LEG_LENGTH = road.geometry().length() / 2
+                    elif (road.geometry().length() - (LEG_LENGTH * 2)) < MAX_SNAP_LENGTH:
+                        LEG_LENGTH = road.geometry().length() / 2
```

Comment:
After getting the UI stuff sorted so I could add a way to define maximum amount a leg is allowed to extend out, the actual logic of using it and modifying the geometry was way easier than I had hoped.